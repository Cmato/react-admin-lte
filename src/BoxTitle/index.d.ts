export { default } from './BoxTitle';
export * from './BoxTitle';
