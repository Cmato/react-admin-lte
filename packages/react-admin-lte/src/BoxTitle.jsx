import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

const BoxTitle = ({ children, className }) => {
  const classes = {
    "box-title": true,
  }

  return (
    <h3 className={classNames(className, classes)}>{children}</h3>
  );
};

BoxTitle.propTypes = propTypes;

export default BoxTitle;
