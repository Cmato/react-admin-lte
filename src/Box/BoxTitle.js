import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxTitle = ({
  children,
  className: classNameProp,
  component: Component,
  ...other
}) => {
  const className = classNames({
    'box-title': true
  }, classNameProp);

  return (
    <Component className={className} {...other}>
      {children}
    </Component>
  );
};

BoxTitle.propTypes = {
  /**
   * The contents of the box title.
   */
  children: PropTypes.node,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
};

BoxTitle.defaultProps = {
  component: 'h3'
};

export default BoxTitle;
