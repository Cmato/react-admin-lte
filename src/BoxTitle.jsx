import React from 'react';

const propTypes = {
  text: React.PropTypes.string,
};

const BoxTitle = ({ text }) => (
  <h3 className="box-title">{text}</h3>
);

BoxTitle.propTypes = propTypes;

export default BoxTitle;
