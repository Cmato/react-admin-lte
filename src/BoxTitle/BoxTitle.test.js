import React from 'react';
import { shallow } from 'enzyme';
import BoxTitle from './BoxTitle';

it('renders default', () => {
  expect(shallow(<BoxTitle><div /></BoxTitle>))
    .toMatchSnapshot();
});

it('renders custom component', () => {
  expect(shallow(<BoxTitle component="h2"><div /></BoxTitle>))
    .toMatchSnapshot();
});
