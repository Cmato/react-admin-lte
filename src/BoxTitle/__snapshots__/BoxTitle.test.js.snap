// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom component 1`] = `
<h2
  className="box-title"
>
  <div />
</h2>
`;

exports[`renders default 1`] = `
<h3
  className="box-title"
>
  <div />
</h3>
`;
